### PR TITLE
[CDAP-7605] - Initial structure for Cask App Store

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds initial structure for Cask Market place
- Adds Configurable Tabs (smart) component which accepts a configuration to render the tabs and its contents
- Adds a Tab (dumb) component to act as skeleton to render the tabs and its contents
- Adds MarketPlaceEntity component to be used across all tabs in Cask Market Place
- Fixes `<base href/>` issue with react-router
- Fixes cdap.html to remove lib.css stylesheet reference.
- Adds ReactStrap library for component use.

### CODE REVIEW

File: cdap-ui/app/cdap/components/ConfigurableTab/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          </TabContents>
+        </Tabs>
+      </div>
+    );
+  }
+}
+const TabConfig = PropTypes.shape({
+  name: PropTypes.string,
+  content: PropTypes.oneOfType([
+    PropTypes.string,
```
Comment (nitinmotgi) [inline cdap-ui/app/cdap/components/ConfigurableTab/index.js]: remove content string

Comment (nitinmotgi) [inline cdap-ui/app/cdap/components/ConfigurableTab/index.js]: replace everywhere for now with &lt;AllContentTabs/&gt;

File: cdap-ui/app/cdap/components/CaskMarketPlace/CaskMarketPlace.less [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    margin-top: 70px;
+
+    .modal-content {
+      border-radius: 0;
+      height: 100%;
+      border: 0;
+
+      .modal-body {
+        padding: 0;
+        height: ~"calc(100% - 51px)"; // 100% - 51px for modal header
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CaskMarketPlace/CaskMarketPlace.less:32]: will this automatically get prefixed?

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/CaskMarketPlace/CaskMarketPlace.less:32]: Good catch! Will fix this.

File: cdap-ui/app/cdap/components/CaskMarketPlace/CaskMarketPlace.less [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        border-bottom: 0;
+
+        h4 {
+          font-weight: normal;
+          span {
+            line-height: 30px;
+          }
+        }
+        .btn-resource-center {
+          background-color: #8A3800;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CaskMarketPlace/CaskMarketPlace.less]: I think we should extract colors to a variable, so it will be easier to manage

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**